CDP4CrossViewEditor/Generator/CrossviewSheetPMeanUtilities.cs

```diff
@@ -0,0 +1,159 @@
+        /// </summary>
+        private const string RedundancyTypeInternal = "Internal";
+
+        /// <summary>
+        /// Hardcoded redundancy.type
+        /// </summary>
+        private const string RedundancyTypeExternal = "External";
+
+        /// <summary>
+        /// Check if P_mean parameter calculation is possible
+        /// </summary>
+        /// <param name="parameters"></param>
+        /// <returns>True if all power related parameters are present, False otherwise</returns>
+        public static bool IsCalculationPossible(List<ParameterOrOverrideBase> parameters)
+        {
+            // Check if all power required parameters are present
+            if (CrossviewSheetConstants.PowerParameters.Except(new[] { "P_peak" }).Except(parameters.Select(p => p.ParameterType.ShortName)).Any())
+            {
+                return false;
+            }
+
+            var pDutyCycle = parameters.Where(p => p != null)
+                .FirstOrDefault(p => p.ParameterType.ShortName.Equals("P_duty_cyc", StringComparison.InvariantCultureIgnoreCase));
+            var pMean = parameters.Where(p => p != null)
+                .FirstOrDefault(p => p.ParameterType.ShortName.Equals("P_mean", StringComparison.InvariantCultureIgnoreCase));
+
+            if (pDutyCycle == null || pMean == null)
+            {
+                return false;
+            }
+
+            // Check if P_duty_cyc && P_mean have the same option dependency
+
+            if (pDutyCycle.IsOptionDependent != pMean.IsOptionDependent)
+            {
+                return false;
+            }
+
+            if (pDutyCycle.IsOptionDependent &&
+                pMean.IsOptionDependent &&
+                pDutyCycle.ValueSets.Select(vs => vs.ActualOption).Except(pMean.ValueSets.Select(vs => vs.ActualOption)).Any())
+            {
+                return true;
+            }
+
+            // Check if P_duty_cyc && P_mean have the same state dependency, both should be
+
+            if (pDutyCycle.StateDependence == null || pMean.StateDependence == null)
+            {
+                return false;
+            }
+
+            if (pDutyCycle.StateDependence.ActualState.Count == 0 || pMean.StateDependence.ActualState.Count == 0)
+            {
+                return false;
+            }
+
+            return !pDutyCycle.StateDependence.ActualState.Except(pMean.StateDependence.ActualState).Any();
+        }
+
+        /// <summary>
+        /// Calculate P_mean parameter value based on multiple parameter values:
+        /// "P_stby", "P_on", "P_duty_cyc", "redundancy.scheme", "redundancy.type", "redundancy.k", "redundancy.n"
+        /// </summary>
+        public static double? ComputeCalculation(string standByValue, string onValue, string dutyCycleValue, string redundancyScheme, string redundancyType, string redundancyK, string redundancyN)
+        {
+            double.TryParse(standByValue, NumberStyles.Any, CultureInfo.InvariantCulture, out var pStandByValue);
```

Comment:
Should we check the returned boolean, so that if the parsing failed we handle it properly? We should probably just return null in that case.